### DISCUSSION TITLE
x-ref seems to be ignored if x-init is also on element

### DISCUSSION THREAD

**Discussion by crohlfs** [alpinejs/alpine#1589] [7. Bugs (archived)]
I've copied the ref demo from the docs, when I add an x-init attribute the ref seems to no longer get set.

Codepen: https://codepen.io/crohlfs/pen/GRWaWmr

**Comment by danddanddand** [comment]
The x-init by itself will create another component scope, so you either should move the x-init to same element as x-data or use alternative strategy for the outer component to access inner component’s element. 

Here is codepen to demonstrate the x-init creating another component scope: https://codepen.io/danddanddand/pen/eYvaGJY

**Reply by crohlfs** [reply]
Interesting, thanks for that explanation, I guess that makes sense for some use cases. I'll move the x-init into the same element as my x-data element.

**Reply by arcanemachine** [reply]
> The x-init by itself will create another component scope, so you either should move the x-init to same element as x-data or use alternative strategy for the outer component to access inner component’s element.

This is pretty confusing behavior tbh. I'm gonna fix up a PR to mention this in the `x-init` section of the docs.

**Reply by KevinBatdorf** [reply]
Feels more like a bug than something to note. I don't think the `x-init` should be scoped if it's nested inside another component, but only when alone (so the behavior would just be the same as adding the missing `x-data` to the el).

**Reply by danddanddand** [reply]
@KevinBatdorf Ya I'm not sure if I'd consider this bug or intended. When it was presented by Caleb I presumed I'd only do this as very small convenience if maybe initializing simple store from server side rendered data (something like `<div x-init="$store.blah.setup( ${val_from_your_templating_lang}, ${anotherOne} )"></div>`) but anything more complicated would likely go into real js and adding x-data would've been fine so hard to figure out goal on this one.

**Reply by danddanddand** [reply]
I'm also curious why codepen I had demoing this stopped working. Hmm...

**Reply by arcanemachine** [reply]
> Feels more like a bug than something to note.

Agreed. It's very unexpected, especially given how the new docs describe its use as a standalone directive.

However, I made a PR for the docs that can be removed if the behavior goes away, but for now, it's a helluva gotcha that should be mentioned IMO. I lost way more time than I should have before finally isolating the problem.
